src/formatter.ts

```diff
@@ -134,20 +134,20 @@ function* formatNode(n: Node, o: Options = {}) {
       yield indent;
       yield '#'.repeat(n.attributes.level || 1);
       yield SPACE;
-      yield* formatChildren(n, no);
+      yield* trimStart(formatChildren(n, no));
       yield* formatAnnotations(n);
       yield NL;
       break;
     }
     case 'paragraph': {
       yield NL;
-      yield indent;
       yield* formatChildren(n, no);
       yield* formatAnnotations(n);
       yield NL;
       break;
     }
     case 'inline': {
+      yield indent;
```

Comment:
@rpaul-stripe what do you think about moving the `indent` from `paragraph` to `inline` here? Felt a little weird, but this solves for the case where there are `inline` elements in a `list` that don't have a `paragraph` for a parent. While still working for the case where there is a `paragraph`.

Reply:
It does feel weird, but I guess it makes sense. I'm not opposed. 👍